parfive/downloader.py

```diff
@@ -372,28 +374,120 @@ def callback(token, future, main_pb):
                                           total=get_http_size(resp))
                     else:
                         file_pb = None
-                    with open(str(filepath), 'wb') as fd:
-                        while True:
-                            chunk = await resp.content.read(chunksize)
-                            if not chunk:
-                                # Close the file progressbar
-                                if file_pb is not None:
-                                    file_pb.close()
-
-                                return str(filepath)
 
-                            # Write this chunk to the output file.
-                            fd.write(chunk)
+                    # setup asyncio queue and writer
+                    queue = asyncio.Queue()
+                    download_workers = []
+                    writer = self.loop.create_task(
+                        self._write_worker(queue, file_pb, filepath))
+
+                    if max_splits and resp.headers.get('Accept-Ranges', None) == "bytes":
+                        content_length = int(resp.headers['Content-length'])
```

Comment:
Some sites might not serve this header right? Should we be resilient to that?

Reply:
MDN web docs example seem to use `Content-length` when talking about `Accept-ranges` header. Same with ranged requests [RFC](https://tools.ietf.org/html/rfc7233). I'll add a check for the worst case scenario.

Reply:
Oh yeah, seems unlikely they would serve one and not the other.